docs/api/auth.md

```diff
@@ -0,0 +1,423 @@
+        var data = JSON.stringify({"password":"your-password"});
+        var xhr = new XMLHttpRequest();
+
+        xhr.addEventListener("readystatechange", function () {
+          if (this.readyState === this.DONE) {
+            console.log(this.responseText);
+          }
+        });
+        ```
+
+    === "Javascript (jQuery)"
+
+        ``` javascript
+        $.ajax({
+          url: "https://pi.hole/admin/api/auth",
+          type: "POST",
+          data: JSON.stringify({"password":"your-password"}),
+          dataType: "json",
+          contentType: "application/json"
+        }).done(function(data) {
+          console.log(data);
+        }).fail(function(xhr, status, error) {
+          console.log(error);
+        });
+        ```
+
+    === "C"
+
+        ``` c
+        #include <stdio.h>
+        #include <stdlib.h>
+        #include <curl/curl.h>
+
+        int main(void)
+        {
+          CURL *curl;
+          CURLcode res;
+
+          curl_global_init(CURL_GLOBAL_ALL);
+
+          curl = curl_easy_init();
+          if(curl) {
+            curl_easy_setopt(curl, CURLOPT_URL, "https://pi.hole/admin/api/auth");
+            curl_easy_setopt(curl, CURLOPT_POSTFIELDS, "{\"password\":\"your-password\"}");
+            curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 0L);
+            curl_easy_setopt(curl, CURLOPT_SSL_VERIFYHOST, 0L);
+            curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "POST");
+
+            struct curl_slist *headers = NULL;
+            headers = curl_slist_append(headers, "Content-Type: application/json");
+            curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers);
+
+            res = curl_easy_perform(curl);
+
+            curl_easy_cleanup(curl);
+          }
+
+          curl_global_cleanup();
+
+          return 0;
+        }
+        ```
+
+    **Parameters**
+
+    Specify either your password or an application password as `password` in the payload.
+
+???+ success "Success response"
+
+    Response code: `HTTP/1.1 200 OK`
+
+    ```json
+    {
+      "session": {
+        "valid": true,
+        "totp": false,
+        "sid": "vFA+EP4MQ5JJvJg+3Q2Jnw=",
+        "csrf": "Ux87YTIiMOf/GKCefVIOMw=",
+        "validity": 300
+      },
+      "took": 0.0002
+    }
+    ```
+
+???+ failure "Error response"
+
+    Response code: `HTTP/1.1 401 - Unauthorized`
+
+    ```json
+    {
+      "error": {
+        "key": "unauthorized",
+        "message": "Unauthorized",
+        "hint": null
+      },
+      "took": 0.003
+    }
+    ```
+
+On success, this will return a JSON object containing the session ID (SID) and the time until your session expires (in seconds).
+You can use this SID from this point to authenticate your requests to the API.
+
+## Use the SID to access API endpoints
+
+Once you have a valid SID, you can use it to authenticate your requests. You can do this in four different ways:
+
+1. In the request URI: `http://pi.hole/admin/api.php?sid=vFA+EP4MQ5JJvJg+3Q2Jnw=`
```

Comment:
I tried various ways to get the sid using method 1 and was never able to do it. In the end, I used a Chrome plugin to modify the header. If the example could return a result, like top domains, it may be more instructive.